service/resource/cloudformation/load_balancer.go

```diff
@@ -0,0 +1,44 @@
+package cloudformation
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/giantswarm/aws-operator/service/key"
+	"github.com/giantswarm/awstpr"
+	"github.com/giantswarm/microerror"
+)
+
+// LoadBalancerName produces a unique name for the load balancer.
+// It takes the domain name, extracts the first subdomain, and combines it with the cluster name.
+func LoadBalancerName(domainName string, cluster awstpr.CustomObject) (string, error) {
+	if key.ClusterID(cluster) == "" {
+		return "", microerror.Maskf(missingCloudConfigKeyError, "spec.cluster.cluster.id")
+	}
+
+	componentName, err := componentName(domainName)
+	if err != nil {
+		return "", microerror.Maskf(malformedCloudConfigKeyError, "spec.cluster.cluster.id")
+	}
+
+	lbName := fmt.Sprintf("%s-%s", key.ClusterID(cluster), componentName)
+
+	return lbName, nil
+}
```

Comment:
I think this should be in the key package.

Reply:
Sure, moved.